### PR TITLE
Dev error analysis

### PR DESCRIPTION
**Pull request by etgld**

### CODE REVIEW

**Comment by etgld** [comment]
Just did a rebase with the dev-v0.7.0 updates from 8/24/23.  At some point need to add back in Wonjin's logic for outputting classification probabilities for predictions but I think that shouldn't be too weird

**Comment by etgld** [comment]
@tmills I think this one is ready to merge.  The column thing with the other issue I created seems to be independent of this

File: src/cnlpt/cnlp_metrics.py
Comment (tmills) [inline src/cnlpt/cnlp_metrics.py:1]: Looks like you are immediately redefining relevant_labels and relevant_preds, can you just delete the original assignments?

File: src/cnlpt/train_system.py [hunk 1]
```diff
@@ -82,6 +108,108 @@ def is_hub_model(model_name: str) -> bool:
     return False
 
 
+def get_dataset_segment(
```
Comment (tmills) [inline src/cnlpt/train_system.py:111]: move to cnlp_data?

File: src/cnlpt/train_system.py [hunk 2]
```diff
@@ -82,6 +108,108 @@ def is_hub_model(model_name: str) -> bool:
...
+):
+    start_ind = end_ind = 0
+    for ind in range(dataset_ind):
+        start_ind += len(dataset.datasets[ind][split_name])
+    end_ind = start_ind + len(dataset.datasets[dataset_ind][split_name])
+
+    return Dataset.from_dict(dataset.processed_dataset[split_name][start_ind:end_ind])
+
+
+def restructure_prediction(
```
Comment (tmills) [inline src/cnlpt/train_system.py:124]: move these 2 functions to cnlp_predict?

File: src/cnlpt/cnlp_metrics.py
Comment (etgld) [reply src/cnlpt/cnlp_metrics.py:1]: Modifying

File: src/cnlpt/train_system.py (hunk 1 above)
Comment (etgld) [reply src/cnlpt/train_system.py]: Moved to cnlp_data, below the definition of ClinicalNlpDataset.  Maybe it should be a member function, along with the functionality for finding the segment, can look into that and make a PR if that looks like a cleaner option

File: src/cnlpt/train_system.py (hunk 2 above)
Comment (etgld) [reply src/cnlpt/train_system.py]: Moved restructure_prediction and structure_labels to cnlp_predict

**Pull request merged**